tests/tensor_search/integ_tests/test_embed.py

```diff
@@ -0,0 +1,438 @@
+
+            tensor_fields=["text_field_1", "text_field_2"]
+        )
+
+        structured_default_image_index = cls.structured_marqo_index_request(
+            model=Model(name='open_clip/ViT-B-32/laion400m_e31'),
+            fields=[
+                FieldRequest(name="text_field_1", type=FieldType.Text,
+                             features=[FieldFeature.LexicalSearch, FieldFeature.Filter]),
+                FieldRequest(name="text_field_2", type=FieldType.Text,
+                             features=[FieldFeature.LexicalSearch, FieldFeature.Filter]),
+                FieldRequest(name="image_field_1", type=FieldType.ImagePointer),
+                FieldRequest(name="image_field_2", type=FieldType.ImagePointer),
+                FieldRequest(name="list_field_1", type=FieldType.ArrayText,
+                             features=[FieldFeature.Filter]),
+            ],
+            tensor_fields=["text_field_1", "text_field_2", "image_field_1", "image_field_2"]
+        )
+
+        structured_image_index_with_random_model = cls.structured_marqo_index_request(
+            model=Model(name='random/small'),
+            fields=[
+                FieldRequest(name="text_field_1", type=FieldType.Text,
+                             features=[FieldFeature.LexicalSearch, FieldFeature.Filter]),
+                FieldRequest(name="text_field_2", type=FieldType.Text,
+                             features=[FieldFeature.LexicalSearch, FieldFeature.Filter]),
+                FieldRequest(name="image_field_1", type=FieldType.ImagePointer)
+            ],
+            tensor_fields=["text_field_1", "text_field_2", "image_field_1"]
+        )
+
+        cls.indexes = cls.create_indexes([
+            unstructured_default_text_index,
+            unstructured_default_image_index,
+            unstructured_image_index_with_random_model,
+            structured_default_text_index,
+            structured_default_image_index,
+            structured_image_index_with_random_model
+        ])
+
+        # Assign to objects so they can be used in tests
+        cls.unstructured_default_text_index = cls.indexes[0]
+        cls.unstructured_default_image_index = cls.indexes[1]
+        cls.unstructured_image_index_with_random_model = cls.indexes[2]
+        cls.structured_default_text_index = cls.indexes[3]
+        cls.structured_default_image_index = cls.indexes[4]
+        cls.structured_image_index_with_random_model = cls.indexes[5]
+
+    def setUp(self) -> None:
+        super().setUp()
+        # Any tests that call add_documents, search, bulk_search need this env var
+        self.device_patcher = mock.patch.dict(os.environ, {"MARQO_BEST_AVAILABLE_DEVICE": "cpu"})
+        self.device_patcher.start()
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        self.device_patcher.stop()
+
+    def test_embed_equivalent_to_add_docs(self):
+        for index in [self.unstructured_default_text_index, self.structured_default_text_index]:
```

Comment:
A single line description for each test would be great. Consider following the naming convention discussed before

Reply:
done